File: test/day2_test.exs
defmodule Day2Test do
  use ExUnit.Case
  require Logger

  @example File.read!("priv/day2/example.txt") |> String.split("\n")
  @input File.read!("priv/day2/input.txt") |> String.split("\n")

  test "part 1" do
    assert Day2.part1(@example) == 15
    Logger.info("Day 2, Part 1: #{Day2.part1(@input)}")
  end

  test "part 2" do
    assert Day2.part2(@example) == 12
    Logger.info("Day 1, Part 2: #{Day2.part2(@input)}")
  end
end

File: kotlin/aoc/src/test/kotlin/y2024/Day2Test.kt
package y2024

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import utils.readFile

class Day2Test {
    private val day = Day2()
    private val file = "day2"

    @Test
    fun part1() {
        assertEquals(2, day.part1(readFile("/y2024/${file}_sample.txt")))
        assertEquals(502, day.part1(readFile("/y2024/${file}.txt")))
    }

    @Test
    fun part2() {
        assertEquals(4, day.part2(readFile("/y2024/${file}_sample.txt")))
        assertEquals(0, day.part2(readFile("/y2024/${file}.txt")))
    }
}
